src/google/cloud/ndb/query.py

```diff
@@ -1022,6 +1061,200 @@ def __init__(
             self._check_properties(self._to_property_names(group_by))
             self.group_by = tuple(group_by)
 
+    def __repr__(self):
+        args = []
+        if self.app is not None:
+            args.append("app=%r" % self.app)
+        if self.namespace is not None:
+            args.append("namespace=%r" % self.namespace)
+        if self.kind is not None:
+            args.append("kind=%r" % self.kind)
+        if self.ancestor is not None:
+            args.append("ancestor=%r" % self.ancestor)
+        if self.filters is not None:
+            args.append("filters=%r" % self.filters)
+        if self.orders is not None:
+            args.append("orders=%r" % self.orders)
+        if self.projection:
+            args.append(
+                "projection=%r" % (self._to_property_names(self.projection))
+            )
+        if self.group_by:
+            args.append(
+                "group_by=%r" % (self._to_property_names(self.group_by))
+            )
+        if self.default_options is not None:
+            args.append("default_options=%r" % self.default_options)
+        return "%s(%s)" % (self.__class__.__name__, ", ".join(args))
+
+    def _get_query(self, client):
```

Comment:
I'm not sure how useful this method will be, ultimately. Currently, as of #44, we're taking the NDB query and generating a  Datastore protocol buffer in `_datastore_query`. I'd leave it for now, though, for reference.